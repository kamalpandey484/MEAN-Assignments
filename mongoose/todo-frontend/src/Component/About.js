import React, { Component } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css'

class About extends Component{

    render() {
        return(
            <div className="AboutPage">
              <h1>About Component!</h1>
            </div>
        );
    }
}
export default About;
