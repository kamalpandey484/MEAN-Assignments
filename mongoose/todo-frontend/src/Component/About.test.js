import React from 'react';
import About from "./About";
import { shallow } from "enzyme";
import renderer from 'react-test-renderer';

const wrapper = shallow(<About />);
let container, containerProp;

describe("FirstComponent - No props", () => {
    test("Is snapshot Same", () => {
        const tree = renderer.create(<About />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    describe("render <div>", () => {
        beforeEach(() => {
            container = wrapper.find("div");
            containerProp = container.props();
        });

        it("should have a <div>", () => {
            expect(container).toHaveLength(1);
        });

        it("should have a <div> with properly className prop", () => {
            expect(containerProp.className).toEqual("AboutPage");
        });
    });

    describe("render <h1>", () => {
        beforeEach(() => {
            container = wrapper.find("h1");
        });

        it("should have a <h1>", () => {
            expect(container).toHaveLength(1);
        });

        it("should have a <h1> with the properly text", () => {
            expect(container.text()).toEqual("About Component!");
        });
    });
});