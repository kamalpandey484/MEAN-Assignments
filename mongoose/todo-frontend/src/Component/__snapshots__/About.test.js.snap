// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstComponent - No props Is snapshot Same 1`] = `
<div
  className="AboutPage"
>
  <h1>
    About Component!
  </h1>
</div>
`;
